src/package_index.rs

```diff
@@ -184,6 +223,32 @@ impl PackageIndex {
         }
     }
 
+    /// Read the list of owners for a scope from the index
+    pub fn get_scope_owners(&self, scope: &str) -> anyhow::Result<Option<Vec<u64>>> {
```

Comment:
This should just return an `anyhow::Result<Vec<u64>>`. I don't think there's a reason to differentiate between "the file didn't exist" and "the file was empty"

Reply:
I was worried that an io error could be mistaken as the file not existing and so the the list of owners could get overwritten or ignored. Maybe I'm just being paranoid though.

Reply:
If we're worried about that, we can make it so the file is not JSON and is instead just a newline delimited list of names. Then, we can set the options for file opening to be append or create. This would make sure this wouldn't happen.

That said, I think having a permission error on read but not on write would be very strange and unlikely.

Reply:
An io error will return an error. That's not related to the cases that return `None` here.

Reply:
Ah yeah I see what you mean! I have simplified this logic now.